activiti-engine/src/main/java/org/activiti/engine/impl/util/ProcessInstanceHelper.java

```diff
@@ -120,13 +120,16 @@ public ProcessInstance createAndStartProcessInstanceByMessage(ProcessDefinition
       throw new ActivitiException("No message start event found for process definition " + processDefinition.getId() + " and message name " + messageName);
     }
 
-    ProcessInstance processInstance = createAndStartProcessInstanceWithInitialFlowElement(processDefinition, businessKey, null, initialFlowElement, process, variables, transientVariables, true);
+    // Create process instance with executions but defer to start process after dispatching ACTIVITY_MESSAGE_RECEIVED
+    ProcessInstance processInstance = createAndStartProcessInstanceWithInitialFlowElement(processDefinition, businessKey, null, initialFlowElement, process, variables, transientVariables, false);
```

Comment:
What about extracting a method `createProcessInstanceWithInitialFlowElement`Instead of calling this one passing `false` for the latest parameter? This method, that can be private, could have `ExecutionEntity` as return type to avoid the cast in line 142. What do you think?

Reply:
Done. See commit 0266f194a3a7e657bfb3ed7e336691b2abd9bff9